src/validation/tests/array_validation_test.rs

```diff
@@ -105,3 +105,176 @@ fn array_access_dimension_mismatch() {
     assert_eq!(diagnostics.len(), 4);
     assert_validation_snapshot!(diagnostics);
 }
+
+#[test]
+fn assignment_1d() {
+    let diagnostics = parse_and_validate(
+        "
+		FUNCTION main : DINT
+			VAR
+				arr 		: ARRAY[1..5] OF DINT := [1, 2, 3, 4, 5, 6];
+				arr_alt		: ARRAY[1..5] OF DINT := (1, 2, 3, 4, 5, 6);
+			END_VAR
+
+			arr := [1, 2, 3, 4, 5]; // Valid
+			arr := [1, 2, 3, 4, 5, 6];
+			arr := (1, 2, 3, 4, 5, 6);
+		END_FUNCTION
+		",
+    );
+
+    assert_validation_snapshot!(diagnostics);
+}
+
+#[test]
+fn assignment_2d() {
+    let diagnostics = parse_and_validate(
+        "
+		FUNCTION main : DINT
```

Comment:
same here, don't test all combinations but a nested array with less elements than needed, it should be valid